eng/Packaging.targets

```diff
@@ -0,0 +1,46 @@
+<Project>
+
+  <ItemGroup>
+    <Content Include="$(RepoRoot)eng\pkg\mlnetlogo.png" Pack="true" PackagePath="" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(IncludeMLNetNotices)' != 'false'">
+    <Content Include="$(RepoRoot)\THIRD-PARTY-NOTICES.TXT" Pack="true" PackagePath="" />
+    <Content Include="$(RepoRoot)\LICENSE" Pack="true" PackagePath=""/>
+  </ItemGroup>
+  
+  <PropertyGroup>    
+    <TargetsForTfmSpecificContentInPackage>$(TargetsForTfmSpecificContentInPackage);IncludeAdditionalFilesInPackage;</TargetsForTfmSpecificContentInPackage>
+  </PropertyGroup>
+
+  <Target Name="_IncludeProjectReferencesWithPackAttributeInPackage"
+          Condition="'@(ProjectReference)' != '' and @(ProjectReference->AnyHaveMetadataValue('Pack', 'true'))"
+          DependsOnTargets="BuildOnlySettings;ResolveReferences">
+    <ItemGroup>
+      <!-- Add ReferenceCopyLocalPaths for ProjectReferences which are flagged as Pack="true" into the package. -->
+      <_projectReferenceCopyLocalPaths Include="@(ReferenceCopyLocalPaths->WithMetadataValue('ReferenceSourceTarget', 'ProjectReference')->WithMetadataValue('Pack', 'true'))" />
+      <TfmSpecificPackageFile Include="@(_projectReferenceCopyLocalPaths)"
+                              PackagePath="$([MSBuild]::ValueOrDefault('%(ReferenceCopyLocalPaths.PackagePath)', '$(BuildOutputTargetFolder)\$(TargetFramework)\'))" />
+    </ItemGroup>
+  </Target>
```

Comment:
Maybe add a comment mentioning that this was copied from runtime. Should we upstream this target into arcade?

Reply:
I simplified this a bit as the runtime version handled RID-specific cases and reference assemblies which we didn't need.  Also split it to share the PDB splitting with the native packaging that's done here.  I can add a comment on where this was referenced from - it's also similar to the targets we already had spread across this repo.

I do think this is very common functionality -- @JeremyKuhne was talking to me the other day that he'd like to use this in Drawing.Common as well.  If you can see a way to add this to arcade, go for it.  Would love to see [this in NuGet/SDK](https://github.com/NuGet/Home/issues/3891).  One of the pitfalls/challenges in making this general purpose is how to do it in a way that can lift the dependencies from the referenced projects (or at least validate they were manually lifted).